### DISCUSSION TITLE
Login failed with PHP 7.2

### DISCUSSION THREAD

**Discussion by IngoBelka** [datenstrom/community#328] [Report a bug] [answered]
Hello Everybody,
I have a fresh installed system with:
* Ubuntu Server 18.04 LTS
* Apache 2.4.29
* PHP 7.2.5
* Yellow 0.7.3
** YellowCommand 0.7.5
** YellowCore 0.7.3
** YellowEdit 0.7.17
** YellowImage 0.7.3
** YellowLanguage 0.7.8
** YellowMarkdown 0.6.8
** YellowUpdate 0.7.10
** YellowThemeFlatsite 0.7.5

I get "Login Failed" even if I could set all the initial data of the fresh installation. I read, that Yellow should set a cookie for a working login, but there comes not a cookie. Setting cookies is allowd for the ip of this server. I have no idea, what I can do now.

I addition to that I get:
`
ERROR checking updates: Plugin 'update' requires cURL library!
`
but dpkg sayes that it is installed```
```
ii  curl                                  7.58.0-2ubuntu3.1                 amd64        command line tool for transferring data with URL syntax
ii  libcurl3-gnutls:amd64                 7.58.0-2ubuntu3.1                 amd64        easy-to-use client-side URL transfer library (GnuTLS flavour)
ii  libcurl4:amd64                        7.58.0-2ubuntu3.1                 amd64        easy-to-use client-side URL transfer library (OpenSSL flavour)
```

Any help is appreciated.

**Comment by IngoBelka** [comment]
Me again,

Apache log file sayes,
```
147.61.10.4 - - [10/Jun/2018:09:56:38 +0200] "GET /yellow/edit/ HTTP/1.1" 200 2861 "http://147.61.10.210/yellow/" "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0"
147.61.10.4 - - [10/Jun/2018:09:56:42 +0200] "POST /yellow/edit/ HTTP/1.1" 430 8842 "http://147.61.10.210/yellow/edit/" "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:60.0) Gecko/20100101 Firefox/60.0"
```

**Comment by schulle4u** [comment]
Have you installed Yellow using the setup wizard provided by the edit plugin or on the command line? 

Regarding Curl: You need the appropriate PHP module in addition to the actual curl libraries. Try installing the php7.2-curl package, (might vary on your system)  this should solve the updater problem. 

HTH
Steffen

**Comment by dirkclemens** [comment]
I just did an update from command line using
`php yellow.php update force`

with the result:

`YellowBlog 0.7.4
YellowSearch 0.7.2
YellowWiki 0.7.4`

and now I can no more edit pages:
`
Login failed

The email or password is incorrect. Please try again.
`

**Comment by Rustapo** [comment]
I have also this issue.

Just make PHP version from 7.2 to 7.1 or 7.0

**Comment by IngoBelka** [comment]
@schulle4u 
"Installing is unzipping one file and you are ready to go. Just give it a try. If you have to read the instructions first, then we have done something wrong." ;) 
Thats what I made, and it is really not trivial to install, without reading instructions, and nevertheless it doesn't work.
The php curl package I installed now, but I had to install the php zip pakage too. I could update the installation then, succesfully.

But a login is not possible.

@Rustapo 
I think the ball is in the field of Datenstrom, to make Yellow working with Php7.2.

**Comment by Rustapo** [comment]
@IngoBelka 
Yes need wait New release of Yellow CMS to working with PHP 7.2 and higher.

But I give suggestion to resolve this issue.

> Login failed
> The email or password is incorrect. Please try again.

**Comment by dirkclemens** [comment]
@Rustapo: works for me, thanks !!!

**Comment by IngoBelka** [comment]
@Rustapo 
I installed PHP7.1 and switched Apache to use it, and I can login. Thank you for the help. Now I'm going to lern, how to work with Yellow. I'll coming back, if I will running into problems.

**Comment by markseuffert** [comment]
After some debugging it looks like something is broken in the PHP 7.2 version that comes with Ubuntu Server 18.04 LTS. Here's a short test case that works fine with previous PHP versions, but it doesn't work with PHP 7.2.5:

Script:

```
<?php
	echo "PHP ".PHP_VERSION." ".PHP_SAPI."<br>\n";

	$token = "ABC";
	$lengthBytes = strlenb($token);
	echo "$token $lengthBytes ".($lengthBytes!=0 ? "ok" : "error")."<br>\n";

	testString();
	function testString()
	{
		$token = "ABC";
		$lengthBytes = strlenb($token);
		echo "$token $lengthBytes ".($lengthBytes!=0 ? "ok" : "error")."<br>\n";
	}
	
	function strlenb() { return call_user_func_array("strlen", func_get_args()); }
?>
```

Result:

```
PHP 7.2.5-0ubuntu0.18.04.1 apache2handler
ABC 3 ok
ABC error
```

As a workaround disable the PHP OpCache.  For example add the following to your `.htaccess` file:

```
php_flag opcache.enable Off
```

**Comment by IngoBelka** [comment]
Hej Mark, I switched back to PhP7.2.x and set 
`php_flag opcache.enable Off`
Now I can login with PHP7.2, too.
While upgrading the system this morning I saw that Version Php7.2.6 came on my system, but this didn't changed the behaviour. So, I'll leave the opcache setting as you suggested.

**Comment by nibreh** [comment]
After install **Yellow v0.7.5** then update online, it seems to work now

After looking at the PHPInfo from my webhost:
https://www.legtux.org/phpinfo.php

**PHP is v7.2.8-1**

If someone can test with the same PHP version, please confirm ..

**Comment by markseuffert** [comment, marked as answer]
Good news, Ubuntu has updated to PHP 7.2.10 which finally includes the bugfix:

~~~
PHP 7.2.10-0ubuntu0.18.04.1 apache2handler
ABC 3 ok
ABC 3 ok
~~~

Should be working now.

**Comment by BsNoSi** [comment]
I can confirm 7.2.10 as working.
